sbi/inference/snpe/snpe_base.py

```diff
@@ -158,18 +161,26 @@ def __call__(
         for round_, num_sims in enumerate(num_sims_per_round):
 
             # Run simulations for the round.
-            theta, x, prior_mask = self._run_simulations(round_, num_sims)
-            x_shape = x_shape_from_simulation(x)
+            theta, x = self._run_simulations(round_, num_sims)
+
+            # Append data to self._theta_bank, self._x_bank, and self._prior_masks.
+            self._append_training_data(
+                theta, x, external_data, round_, exclude_invalid_x
```

Comment:
why do we need to pass the external data here? Couldn't we handle this internally, e.g., in `run_simulations`? E.g., introduce it to all methods (-> to base.py) and rename it to `get_simulations` or so and use it to merge potential presimulated data?